userspace/plugin/plugin_types.h

```diff
@@ -293,6 +293,37 @@ typedef enum ss_plugin_log_severity
 	SS_PLUGIN_LOG_SEV_TRACE = 8,
 } ss_plugin_log_severity;
 
+// Types supported by the by the metric values
+typedef enum ss_plugin_metric_value_type
```

Comment:
Would it be possible to explore ways to not redefine all this? Since it's already defined in `metrics_v2.h`?
I also don't believe we need plugin specific metric types. That way we save the casting of types back and forth.

Plugin metrics could be as simple as just a new category using the existing libs metrics definitions. This will also make it easier to maintain in the long run.

Reply:
I think we need to redefine those types, considering that we need the plugin API to be separate from the inspector code.
If we really want to avoid redefining, we can make the inspector use the types defined in the plugin API and remove the ones from `metrics_v2.h`, but I don't know if that's ideal.

Reply:
See https://github.com/falcosecurity/libs/pull/1828#issuecomment-2103224934